PhysicsTools/NanoAOD/plugins/BToKsteeProducer.cc

```diff
@@ -0,0 +1,790 @@
+
+			  math::XYZVector refitKaonV3D_Kst = refitKaon_Kst->refittedTransientTrack().track().momentum();
+			  math::XYZVector refitPionV3D_Kst = refitPion_Kst->refittedTransientTrack().track().momentum();
+			  refitKstV3D = refitKaonV3D_Kst + refitPionV3D_Kst;
+
+			}
+
+		      }                    
+		      
+		      RefCountedKinematicVertex refitVertexBToKstEE;
+		      RefCountedKinematicParticle refitBToKstEE;
+		      RefCountedKinematicParticle refitEle1;
+		      RefCountedKinematicParticle refitEle2;
+		      RefCountedKinematicParticle refitKaon;
+		      RefCountedKinematicParticle refitPion;                    
+                    
+		      bool passed = BToKstEEVertexRefitting(ele1, ele2, kaon, pion,
+							    theTTBuilder,
+							    refitVertexBToKstEE,
+							    refitBToKstEE,
+							    refitEle1,
+							    refitEle2,
+							    refitKaon,
+							    refitPion);
+                    
+		      if (!passed) continue;
+                    
+		      pair<double,double> BToKstEELS = computeLS(refitVertexBToKstEE,beamSpot);
+		      double LSBS = BToKstEELS.first;
+		      double LSBSErr = BToKstEELS.second;
+                    
+		      double BToKstEEVtx_CL = TMath::Prob((double)refitVertexBToKstEE->chiSquared(),
+							  int(rint(refitVertexBToKstEE->degreesOfFreedom())));
+
+                    
+		      double cosAlpha = computeCosAlpha(refitBToKstEE,refitVertexBToKstEE,beamSpot);
+                    
+		      double mass_err = sqrt(refitBToKstEE->currentState().kinematicParametersError().matrix()(6,6));              
+      
+		      pat::CompositeCandidate BToKstEECand;
+		      BToKstEECand.addDaughter( ele1 , "ele1");
+		      BToKstEECand.addDaughter( ele2 , "ele2");
+		      BToKstEECand.addDaughter( kaon, "kaon");
+		      BToKstEECand.addDaughter( pion, "pion");
+
+		      BToKstEECand.addUserInt("ele1_index", i);
+		      BToKstEECand.addUserInt("ele2_index", j);
+		      BToKstEECand.addUserInt("kaon_index", k);
+		      BToKstEECand.addUserInt("pion_index", l);
+		      
+		      math::XYZVector refitEle1V3D = refitEle1->refittedTransientTrack().track().momentum();
+		      BToKstEECand.addUserFloat("ele1_pt",     sqrt(refitEle1V3D.perp2()));
+		      BToKstEECand.addUserFloat("ele1_eta",    refitEle1V3D.eta());
+		      BToKstEECand.addUserFloat("ele1_phi",    refitEle1V3D.phi());
+		      BToKstEECand.addUserFloat("ele1_charge", refitEle1->currentState().particleCharge());
+
+		      math::XYZVector refitEle2V3D = refitEle2->refittedTransientTrack().track().momentum();
+		      BToKstEECand.addUserFloat("ele2_pt",     sqrt(refitEle2V3D.perp2()));
+		      BToKstEECand.addUserFloat("ele2_eta",    refitEle2V3D.eta());
+		      BToKstEECand.addUserFloat("ele2_phi",    refitEle2V3D.phi());
+		      BToKstEECand.addUserFloat("ele2_charge", refitEle2->currentState().particleCharge());		    
+
+		      TLorentzVector ele1cand;
```

Comment:
here the logic is ok, but with eta, pt, phi of the electrons from the vtx fit of refitKst, ele1, ele2
instead of the 4 independent particles